lib/ex_aws/cloud_search/document.ex

```diff
@@ -0,0 +1,5 @@
+defprotocol ExAws.CloudSearch.Document do
+  @fallback_to_any true
```

Comment:
Well if there are cases where I may have some `%User{}` and want to provide some customization with respect to the way in which it's encoded, I think a protocol is fine. I'm not really all that familiar with cloudsearch yet so it isn't entirely clear if such a situation is what we want.

The particular comment here was less about the use of a protocol in and of itself, and more about whether or not we want to fall back to any vs enforce particular implementations to exist.